IPython/html/static/tree/js/newnotebook.js

```diff
@@ -85,16 +85,20 @@ define([
                     url += "?kernel_name=" + kernel_name;
                 }
                 w.location = url;
-            },
-            function (error) {
-                w.close();
-                dialog.modal({
-                    title : 'Creating Notebook Failed',
-                    body : "The error was: " + error.message,
-                    buttons : {'OK' : {'class' : 'btn-primary'}}
-                });
-            }
-        );
+        }).catch(function (e) {
```

Comment:
MDN says they're the same- "The catch() method returns a Promise and deals with rejected cases only. It behaves the same as calling Promise.prototype.then(undefined, onRejected)."  Although, looking closer, using .catch after the .then will additionally handle rejections from the preceding then...  More clearly said `P.then(s).catch(e)`, `e` will handle `s`'s rejections in addition to `P`'s whereas with `P.then(s, e)` it's not clear that `e` will handle `s`'s rejection at all (if s is a promise itself).